File: python/authentication.py
def authenticate(users):
    username = input("Enter username: ").strip()
    password = input("Enter password: ").strip()
    
    user = users.get(username)
    if user and user["password"] == password:
        print(f"Welcome, {username}!")
        return username, user["role"]
    else:
        print("Invalid credentials. Please try again.")
        return None, None

if __name__ == "__main__":
    test_users = {
        "admin": {"password": "adminpass", "role": "admin"},
        "student101": {"password": "studpass", "role": "student"}
    }
    authenticate(test_users)

File: python/kotlin-app/app/src/main/kotlin/Authentication.kt
object Authentication {
    data class User(val password: String, val role: String)

    fun authenticate(users: Map<String, User>): Pair<String?, String?> {
        print("Enter username: ")
        val username = readlnOrNull()?.trim() ?: ""
        print("Enter password: ")
        val password = readlnOrNull()?.trim() ?: ""

        val user = users[username]
        return if (user != null && user.password == password) {
            println("Welcome, $username!")
            Pair(username, user.role)
        } else {
            println("Invalid credentials. Please try again.")
            Pair(null, null)
        }
    }
}